#!/usr/bin/env python
PACKAGE = "depth_cam_extrinsics_calib"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("h_min",    int_t,    0, "H min", 0,  0, 360)
gen.add("h_max",    int_t,    0, "H max", 360,  0, 360)

gen.add("s_min",    double_t,    0, "S min", 0,  0, 1.0)
gen.add("s_max",    double_t,    0, "S max", 1.0,  0, 1.0)

gen.add("v_min",    double_t,    0, "V min", 0,  0, 1.0)
gen.add("v_max",    double_t,    0, "V max", 1.0,  0, 1.0)

gen.add("use_StatisticalOutlierRemoval",   bool_t,   0, "Use StatisticalOutlierRemoval",  False)

gen.add("mean_k",    int_t,    0, "Mean K for StatisticalOutlierRemoval", 50,  0, 2000)

exit(gen.generate(PACKAGE, "depth_cam_extrinsics_calib", "calib"))